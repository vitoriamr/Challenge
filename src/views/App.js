import React, { useState } from "react";
import styled from "styled-components";

import { STATES, DEFAULT_VALUE } from "../constants/sampleData";
import Dropdown from "./Dropdown/Dropdown";
import FAQArea from "./FAQArea/FAQArea";
import { getDropdownOptions } from "./Dropdown/dropdownConfig/dropdownConfig";

const App = () => {
  const [firstDropdownCurrentValue, setFirstDropdownCurrentValue] = useState(
    DEFAULT_VALUE
  );
  const [secondDropdownCurrentValue, setSecondDropdownCurrentValue] = useState(
    DEFAULT_VALUE
  );
  const [dropdownAvailable, setDropdownAvailable] = useState(false);
  const [faqVisible, setFaqVisible] = useState(false);

  const dropdown1 = [STATES.NY, STATES.CA, STATES.TX];
  const dropdown2 = dropdownAvailable
    ? getDropdownOptions(firstDropdownCurrentValue)
    : [];

  const handleFirstDropdown = (event, value) => {
    event.preventDefault();
    setFirstDropdownCurrentValue(value);
    setDropdownAvailable(true);
    setSecondDropdownCurrentValue(DEFAULT_VALUE);
    setFaqVisible(false);
  };

  const handleSecondDropdown = (event, value) => {
    event.preventDefault();
    setSecondDropdownCurrentValue(value);
    setFaqVisible(true);
  };

  return (
    <Grid>
      <FlexDiv>
        <DropdownLabel id="statesLabel">{"States:"}</DropdownLabel>
        <Dropdown
          available={true}
          currentValue={firstDropdownCurrentValue}
          valueOptions={dropdown1}
          handleDropdown={handleFirstDropdown}
        />
      </FlexDiv>
      <FlexDiv>
        <DropdownLabel id="citiesLabel">{"Cities:"}</DropdownLabel>
        <Dropdown
          available={dropdownAvailable}
          currentValue={secondDropdownCurrentValue}
          valueOptions={dropdown2}
          handleDropdown={handleSecondDropdown}
        />
      </FlexDiv>
      <FAQArea faqVisible={faqVisible} value={secondDropdownCurrentValue} />
    </Grid>
  );
};

const Grid = styled.div`
  margin: 30px;
  display: grid;
`;

const FlexDiv = styled.div`
  display: flex;
  margin-bottom: 40px;
`;

const DropdownLabel = styled.label`
  margin: 3px 10px 0 0;
`;
DropdownLabel.displayName = "DropdownLabel";

export default App;
