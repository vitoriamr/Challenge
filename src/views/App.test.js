import React from "react";
import { shallow } from "enzyme";

import App from "./App";

describe("App", () => {
  it("matches previous snapshot", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render States label", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('DropdownLabel[id="statesLabel"]').text()).toBe(
      "States:"
    );
  });

  it("should render Cities label", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('DropdownLabel[id="citiesLabel"]').text()).toBe(
      "Cities:"
    );
  });

  it("should render 2 Dropdowns label", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("Dropdown").length).toBe(2);
  });

  it("should render FAQArea", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("FAQArea").exists()).toBe(true);
  });
});
