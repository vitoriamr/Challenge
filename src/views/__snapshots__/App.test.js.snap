// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches previous snapshot 1`] = `
<styled.div>
  <styled.div>
    <DropdownLabel
      id="statesLabel"
    >
      States:
    </DropdownLabel>
    <Dropdown
      available={true}
      currentValue="Select an Option..."
      handleDropdown={[Function]}
      valueOptions={
        Array [
          "New York",
          "California",
          "Texas",
        ]
      }
    />
  </styled.div>
  <styled.div>
    <DropdownLabel
      id="citiesLabel"
    >
      Cities:
    </DropdownLabel>
    <Dropdown
      available={false}
      currentValue="Select an Option..."
      handleDropdown={[Function]}
      valueOptions={Array []}
    />
  </styled.div>
  <FAQArea
    faqVisible={false}
    value="Select an Option..."
  />
</styled.div>
`;
